File: room-manager/src/main/kotlin/org/rivero/roommanager/configuration/IAuthenticationFacade.java
package org.rivero.roommanager.configuration;

import org.springframework.security.core.Authentication;
import org.springframework.security.oauth2.jwt.Jwt;
import reactor.core.publisher.Mono;

import java.security.Principal;

public interface IAuthenticationFacade {
    Mono<Authentication> getAuthentication();

    Mono<Jwt> getPrincipal();
}

File: room-manager/src/main/kotlin/org/rivero/roommanager/configuration/IAuthenticationFacade.kt
package org.rivero.roommanager.configuration

import org.rivero.roommanager.user.AuthUser
import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.jwt.Jwt
import reactor.core.publisher.Mono

interface IAuthenticationFacade {
    val authentication: Mono<Authentication>

    val principal: Mono<Jwt>

    val user: Mono<AuthUser>
}
